fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/ContainmentIndexImpl.java

```diff
@@ -591,4 +660,18 @@ public void setDataSource(final DataSource dataSource) {
         this.dataSource = dataSource;
     }
 
+    /**
+     * Format an instant to a timestamp without milliseconds, due to precision
+     * issues with memento datetimes.
+     * @param instant
+     * @return the timestamp
+     */
+    private Timestamp formatInstant(final Instant instant) {
+        if (instant == null) {
+            return null;
+        }
+        final var timestamp = Timestamp.from(instant);
+        timestamp.setNanos(0);
```

Comment:
Since mementos have second level granularity, perhaps it would make sense to truncate to the second?

Reply:
Is that not what this is doing? Setting the nanoseconds to 0? Perhaps @bbpennel could help as I stole this from [his code](https://github.com/fcrepo/fcrepo/blob/main/fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/MembershipIndexManager.java#L548-L558)

Reply:
I see. It reads as if it's truncating to milliseconds, but I see now that it's just removing all fractions of a second.